### ISSUE TITLE
Fatal Error when using scroll wheel/mouse buttons

### ISSUE THREAD

**Issue by a-lex-ant** [JFormDesigner/FlatLaf#74]
Hello,
I'm encountering a fatal JRE error every time I try to use my mouse's scroll wheel and/or click on anything that is not a button.
It happens with both custom and standard themes.
You can see the error log attached below.
[hs_err_pid16130.log](https://github.com/JFormDesigner/FlatLaf/files/4270207/hs_err_pid16130.log)

Thank you!

**Comment by DevCharly** [comment]
Thanks for reporting.

Can you please try whether this also occur with the MetalLookAndFeel?

Also try disabling FlatLaf smooth scrolling with following code:
~~~java
UIManager.put( "ScrollPane.smoothScrolling", false );
~~~

Looking at the log it seems that you're running your app with the JetBrains JRE, right?
Would be also interesting to know whether this occur with non-JetBrains JRE.

Does the FlatLaf Demo crash too?
https://github.com/JFormDesigner/FlatLaf#demo

Please also try FlatLaf 0.28-SNAPSHOT (see https://github.com/JFormDesigner/FlatLaf#snapshots), which has some major changes in the LaF initialization. Maybe this makes a difference...


Sorry for asking so many questions, but this is hard to reproduce and I like to isolate the source of the problem. Thanks.

**Comment by a-lex-ant** [comment]
Thank you so much Karl, I managed to fix the issue by disabling the smoothScrolling option.

I will answer your other questions in case you are interested:
The issue does not occur with the Metal Look and Feel, nor with any other default Java LaF.
The FlatLaf Demo does not crash and works flawlessly.
I tried to reproduce the error using the OpenJDK Runtime Environment (11.0.6+10) (build 11.0.6+10-post-Ubuntu-1ubuntu119.10.1) by starting my program via terminal outside of Intellij, and it crashed anyway. Same behaviour with FlatLaf 0.28-SNAPSHOT too.

Thank you again for your helpful tips,
Anna

**Comment by DevCharly** [comment]
Hey Anna,

many thanks for your feedback and for trying out all variations...

Great that it works now, but I still wonder why your app crashes with smooth scrolling enabled, but the FlatLaf Demo does not.

Maybe this has something to do with vlcj.
Could you try whether your app crashes with latest version of vlcj 4.3.0 and smooth scrolling enabled?

**Comment by a-lex-ant** [comment]
> Could you try whether your app crashes with latest version of vlcj 4.3.0 and smooth scrolling enabled?
I've been using version 4.3.0 already

> Maybe this has something to do with vlcj.
I think you're right, because I've just tried to remove all the classes that use vlcj and nothing crashes, even with smooth scrolling enabled.
Also, I noticed that the problematic frame comes from native code, and Vlcj does use native libraries, if I remember correctly...

Maybe I should open an issue with them, then?

Thanks :+1:

**Comment by DevCharly** [comment]
There is already a (closed) vlcj issue, where it crashes also in `sun.awt.X11.XToolkit.initializeMultiClickTime`: https://github.com/caprica/vlcj/issues/477

Looking at the comments, I don't have the impression that it was fixed.
The first comment says that it crashes when clicking an empty `java.awt.Frame`, but works with `javax.swing.JFrame`.
The last comment suggest to change the order of initialization:
https://github.com/caprica/vlcj/issues/477#issuecomment-327026914

> Maybe I should open an issue with them, then?

Yes, please.

**Comment by a-lex-ant** [comment]
> The first comment says that it crashes when clicking an empty java.awt.Frame
Yeah, that happens to me too.
> The last comment suggest to change the order of initialization
Unfortunately, that solves nothing for me :(

I'll go on and open the issue with them :+1: 
Thanks!

**Comment by caprica** [comment]
I did some looking into this.

There's nothing new to report really, but:

1. I was able to easily reproduce this if the LookAndFeel was initialised before creating the media player factory
2. I was trivially able to prevent the crash by making sure the media player factory was initialised before the LookAndFeel

There is clearly some initialisation race in native code somewhere, but digging into and debugging native code is not on my agenda especially when it is possible to workaround the problem with some reordering of initialisation code.

Having said that, if I could find a way in vlcj to prevent this happening, that would be useful.
